theano/gpuarray/c_code/spatialtf_sampler.c

```diff
@@ -0,0 +1,220 @@
+#section support_code
+
+typedef struct __spatialtf_context {
+    cudnnTensorDescriptor_t xdesc;
+    cudnnTensorDescriptor_t ydesc;
+} spatialtf_context_t;
+
+void spatialtf_context_init( spatialtf_context_t * ctx )
+{
+    ctx->xdesc = NULL;
+    ctx->ydesc = NULL;
+}
+
+void spatialtf_context_destroy( spatialtf_context_t * ctx )
+{
+    if ( NULL != ctx->xdesc )
+        cudnnDestroyTensorDescriptor( ctx->xdesc );
+
+    if ( NULL != ctx->ydesc )
+        cudnnDestroyTensorDescriptor( ctx->ydesc );
+}
+
+#section support_code_struct
+
+int
+spatialtf_sampler(PyGpuArrayObject * input,
+                  PyGpuArrayObject * grid,
+                  PyArrayObject * grid_dimensions,
+                  cudnnSpatialTransformerDescriptor_t desc,
+                  double alpha, double beta,
+                  PyGpuArrayObject ** output,
+                  cudnnHandle_t _handle)
+{
+    PyGpuContextObject * gpu_ctx = input->context;
+    void * alpha_p;
+    void * beta_p;
+    float af = alpha, bf = beta;
+    spatialtf_context_t spatialtf_ctx;
+    cudnnDataType_t dt;
+    // Number of color channels (feature maps) is the innermost dimension
```

Comment:
That comment should have been removed, sorry about that. In this case, we assume that the width is the innermost dimension, because of the NCHW data layout.